cdap-ui/cypress/integration/plugins.validation.spec.ts

```diff
@@ -0,0 +1,104 @@
+  serviceFilePath: DEFAULT_GCP_SERVICEACCOUNT_PATH,
+};
+
+let headers = {};
+
+describe('Plugin properties', () => {
+  // Uses API call to login instead of logging in manually through UI
+  before(() => {
+    loginIfRequired().then(() => {
+      cy.getCookie('CDAP_Auth_Token').then((cookie) => {
+        if (!cookie) {
+          return;
+        }
+        headers = {
+          Authorization: 'Bearer ' + cookie.value,
+        };
+      });
+    });
+    cy.visit('/pipelines/ns/default/studio');
+  });
+
+  beforeEach(() => {
+    getArtifactsPoll(headers);
+  });
+
+  it('should be validated and errors found in properties', () => {
+    cy.add_node_to_canvas(sourceNode);
+    cy.get(
+          '[data-cy="plugin-node-BigQueryTable-batchsource-0"] .node .node-configure-btn')
+        .invoke('show')
+        .click();
+    // Testing for errors found, rgb(164, 4, 3) is the red border color
+    cy.get('[data-cy="plugin-properties-validate-btn"]').click();
+    cy.get('[data-cy="plugin-properties-errors-found"]')
+        .should('have.text', '3 errors found.');
+    cy.get('[data-cy="referenceName"] > [data-cy="widget-wrapper-container"]')
+        .should('have.css', 'border-color', 'rgb(164, 4, 3)');
+    // Testing if the widget is highlighted when there is an error
+    // and the error text
+    cy.get('[data-cy="referenceName"] ~ [data-cy="property-row-error"]')
+        .should(
+            'have.text', 'Required property \'referenceName\' has no value.');
+    cy.get('[data-cy="referenceName"] > [data-cy="widget-wrapper-container"]')
+        .should('have.css', 'border-color', 'rgb(164, 4, 3)');
+  });
+
+  it('should be validated and no errors found in plugin properties', () => {
+    // Testing for no errors found
+    cy.get('input[data-cy="referenceName"]').type('randomName');
+    cy.get('input[data-cy="dataset"]').type('randomDataset');
+    cy.get('input[data-cy="table"]').type('randomTableName');
+    cy.get('[data-cy="plugin-properties-validate-btn"]').click();
+    cy.get('[data-cy="plugin-properties-no-errors"]')
+        .should('have.text', 'No errors found.');
+  });
+
+  it('should be validated and errors found in schema', () => {
+    // configuring properties for the plugin
+    cy.get('input[data-cy="referenceName"]')
+        .clear()
+        .type(sourceProperties.referenceName);
+    cy.get('input[data-cy="project"]').clear().type(sourceProperties.project);
+    cy.get('input[data-cy="datasetProject"]')
+        .clear()
+        .type(sourceProperties.datasetProject);
+    cy.get('input[data-cy="dataset"]').clear().type(sourceProperties.dataset);
+    cy.get('input[data-cy="table"]').clear().type(sourceProperties.table);
+    cy.get('input[data-cy="serviceFilePath"]')
+        .clear()
+        .type(sourceProperties.serviceFilePath);
+    // retrieving schema
+    cy.get('[data-cy="get-schema-btn"]').click();
+    cy.get('[data-cy="get-schema-btn"]').contains('Get Schema', {
```

Comment:
nit: May be give a higher timeout. Or better yet remove the timeout and have the default timeout for `cy.get` to be like 1min.

Reply:
Fixed